openpype/hosts/max/api/preview_animation.py

```diff
@@ -0,0 +1,310 @@
+        filepath, start, end, ext, viewport_options):
+    """Function to set up preview arguments in MaxScript.
+    ****For 3dsMax 2024+
+    Args:
+        filepath (str): filepath for render output without frame number and
+            extension, for example: /path/to/file
+        start (int): startFrame
+        end (int): endFrame
+        viewport_options (dict): viewport setting options, e.g.
+            {"vpStyle": "defaultshading", "vpPreset": "highquality"}
+    Returns:
+        list: Created files
+    """
+    filepath = filepath.replace("\\", "/")
+    preview_output = f"{filepath}..{ext}"
+    frame_template = f"{filepath}.{{:04d}}.{ext}"
+    job_args = list()
+    default_option = f'CreatePreview filename:"{preview_output}"'
+    job_args.append(default_option)
+    frame_option = f"outputAVI:false start:{start} end:{end}"
+    job_args.append(frame_option)
+    for key, value in viewport_options.items():
+        if isinstance(value, bool):
+            if value:
+                job_args.append(f"{key}:{value}")
+        elif isinstance(value, str):
+            if key == "vpStyle":
+                if value == "Realistic":
+                    value = "defaultshading"
+                elif value == "Shaded":
+                    log.warning(
+                        "'Shaded' Mode not supported in "
+                        "preview animation in Max 2024.\n"
+                        "Using 'defaultshading' instead.")
+                    value = "defaultshading"
+                elif value == "ConsistentColors":
+                    value = "flatcolor"
+                else:
+                    value = value.lower()
+            elif key == "vpPreset":
+                if value == "Quality":
+                    value = "highquality"
+                elif value == "Customize":
+                    value = "userdefined"
+                else:
+                    value = value.lower()
+            job_args.append(f"{key}: #{value}")
+    auto_play_option = "autoPlay:false"
+    job_args.append(auto_play_option)
+    job_str = " ".join(job_args)
+    log.debug(job_str)
+    rt.completeRedraw()
+    rt.execute(job_str)
+    # Return the created files
+    return [frame_template.format(frame) for frame in range(start, end + 1)]
+
+
+def _render_preview_animation_max_pre_2024(
+        filepath, startFrame, endFrame, percentSize, ext):
+    """Render viewport animation by creating bitmaps
+    ***For 3dsMax Version <2024
+    Args:
+        filepath (str): filepath without frame numbers and extension
+        startFrame (int): start frame
+        endFrame (int): end frame
+        percentSize (int): percentage of the resolution
+        ext (str): image extension
+    Returns:
+        list: Created filepaths
+    """
+    # get the screenshot
+    resolution_percentage = float(percentSize) / 100
+    res_width = rt.renderWidth * resolution_percentage
+    res_height = rt.renderHeight * resolution_percentage
```

Comment:
Shouldn't this still be forced to integers?

Reply:
I would use the function you recommended above(It seems more accurate with rounding up the indice and make them integers)